File: typescript/fileresult.ts
export class FileResult{
    filepath: string;
    hash: string;
    size: number;
    date: number;

    constructor(filepath: string, filehash: string, size: number, date: number){
        this.filepath = filepath
        this.hash = filehash
        this.size = size
        this.date = date
    }

    public equals(another: FileResult) : boolean{
        return this.hash == another.hash
            && this.filepath == another.filepath
            && this.size == another.size
            && (this.date - another.date) < 1000
    }

    public toString() : string{
        const dateObject = new Date(this.date)
        const dateString = dateObject.toISOString().split('T')
        return `${this.filepath} (${dateString[0]} ${dateString[1].split('.')[0]} | ${this.size} bytes)`
    }
}

File: csharp/FileResult.cs
using System;
using System.IO;

namespace LanguageBenchmark
{
    public class FileResult
    {
        public string FilePath { get; set; }

        public string HashValue { get; set; }

        public long Size { get; set; }

        public DateTime ModifiedDate { get; set; }

        public override bool Equals(object obj)
        {
            //
            // See the full list of guidelines at
            //   http://go.microsoft.com/fwlink/?LinkID=85237
            // and also the guidance for operator== at
            //   http://go.microsoft.com/fwlink/?LinkId=85238
            //
            var other = obj as FileResult;
            if (other == null)
            {
                return false;
            }
            
            return Path.Equals(this.FilePath, other.FilePath)
                && this.HashValue == other.HashValue
                && this.Size == other.Size
                && Math.Abs((this.ModifiedDate - other.ModifiedDate).TotalSeconds) < 1;
        }

        public override int GetHashCode(){ return base.GetHashCode(); }

        public override string ToString()
        {
            return $"{this.FilePath} ({this.ModifiedDate.ToLongTimeString()} | {this.Size} bytes)";
        }
    }
}